benchmark/tpch-dbaas-v4.0.md

```diff
@@ -0,0 +1,134 @@
+
+```bash
+tiup bench tpch prepare \
+    --host ${tidb_host} --port 4000 --db tpch_10 --password ${password} \
+    --sf 10 \
+    --tiflash \
+    --analyze --tidb_build_stats_concurrency 8 --tidb_distsql_scan_concurrency 30
+```
+
+运行完之后，确认数据已经建立好 TiFlash 副本：
+
+```sql
+select * from information_schema.tiflash_replica;
++--------------+------------+----------+---------------+-----------------+-----------+----------+
+| TABLE_SCHEMA | TABLE_NAME | TABLE_ID | REPLICA_COUNT | LOCATION_LABELS | AVAILABLE | PROGRESS |
++--------------+------------+----------+---------------+-----------------+-----------+----------+
+| tpch_10      | nation     |       53 |             1 |                 |         1 |        1 |
+| tpch_10      | part       |       59 |             1 |                 |         1 |        1 |
+| tpch_10      | supplier   |       62 |             1 |                 |         1 |        1 |
+| tpch_10      | partsupp   |       65 |             1 |                 |         1 |        1 |
+| tpch_10      | region     |       56 |             1 |                 |         1 |        1 |
+| tpch_10      | orders     |       72 |             1 |                 |         1 |        1 |
+| tpch_10      | customer   |       68 |             1 |                 |         1 |        1 |
+| tpch_10      | lineitem   |       75 |             1 |                 |         1 |        1 |
++--------------+------------+----------+---------------+-----------------+-----------+----------+
+```
+
+### 执行查询
+
+本文会比较智能选择、TiKV 隔离读、TiFlash 隔离读等三种[读取模式](/reference/tiflash/use-tiflash.md#使用-tidb-读取-tiflash)下的 TiDB 性能。使用 MySQL 客户端连接到 TiDB 集群，先[优化相关参数](/reference/tiflash/tune-performance.md#tidb-相关参数调优)以及读取的模式。
+
+```sql
+set @@session.tidb_allow_batch_cop = 1;
+set @@tidb_opt_agg_push_down = 1;
+set @@session.tidb_opt_distinct_agg_push_down = 1;
+set @@tidb_distsql_scan_concurrency = 30;
+-- 设置读取模式. 不同读取模式对应的值如下:
+-- 智能选择:'tikv,tiflash', TiKV隔离读:'tikv', TiFlash隔离读:'tiflash'.
+set @@session.tidb_isolation_read_engines = 'tikv,tiflash';
+-- 可以通过以下语句确认当前变量值
+select @@tidb_isolation_read_engines , @@tidb_allow_batch_cop , @@tidb_opt_distinct_agg_push_down , @@tidb_distsql_scan_concurrency;
++-------------------------------+------------------------+-----------------------------------+---------------------------------+
+| @@tidb_isolation_read_engines | @@tidb_allow_batch_cop | @@tidb_opt_distinct_agg_push_down | @@tidb_distsql_scan_concurrency |
++-------------------------------+------------------------+-----------------------------------+---------------------------------+
+| tikv,tiflash                  | 1                      | 1                                 | 30                              |
++-------------------------------+------------------------+-----------------------------------+---------------------------------+
+```
```

Comment:
How about this?
- Add a `copyable` code snippet
- Separate the command and the result
---

{{< copyable "sql" >}}

```sql
set @@session.tidb_allow_batch_cop = 1;
set @@tidb_opt_agg_push_down = 1;
set @@session.tidb_opt_distinct_agg_push_down = 1;
set @@tidb_distsql_scan_concurrency = 30;
-- 设置读取模式. 不同读取模式对应的值如下:
-- 智能选择:'tikv,tiflash', TiKV隔离读:'tikv', TiFlash隔离读:'tiflash'.
set @@session.tidb_isolation_read_engines = 'tikv,tiflash';
-- 可以通过以下语句确认当前变量值
select @@tidb_isolation_read_engines , @@tidb_allow_batch_cop , @@tidb_opt_distinct_agg_push_down , @@tidb_distsql_scan_concurrency;
```

```
+-------------------------------+------------------------+-----------------------------------+---------------------------------+
| @@tidb_isolation_read_engines | @@tidb_allow_batch_cop | @@tidb_opt_distinct_agg_push_down | @@tidb_distsql_scan_concurrency |
+-------------------------------+------------------------+-----------------------------------+---------------------------------+
| tikv,tiflash                  | 1                      | 1                                 | 30                              |
+-------------------------------+------------------------+-----------------------------------+---------------------------------+
```